transport/x/grpc/peer.go

```diff
@@ -159,7 +160,17 @@ func (p *grpcPeer) wait() error {
 	if p.stopped {
 		return p.stoppedErr
 	}
-	p.stoppedErr = <-p.stoppedC
+
+	// TODO Fix in another PR
+	// Stopping the outbound always timed out at the stoppedC in my tests, this
+	// is just a workaround that doesn't solve the core issue.
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
```

Comment:
Yeah, this was why I think there's a race condition in the GRPC peer API, whenever I would stop the grpc outbound the p.stoppedC channel would never close and block indefinitely